test/e2e/density.go

```diff
@@ -562,6 +587,45 @@ var _ = framework.KubeDescribe("Density", func() {
 			}
 		})
 	}
+
+	It("should allow running maximum capacity pods on nodes", func() {
+		// Calculate total number of pods from each node's max-pod
+		totalPods = 0
+		for _, n := range nodes.Items {
```

Comment:
This won't work, as you need to take into account Pods already running in the cluster (e.g. KubeProxy on each node). Take a look at SchedulerPredicates test.